src/hbbft.erl

```diff
@@ -432,6 +432,61 @@ deserialize(#hbbft_serialized_data{batch_size=BatchSize,
                                    stampfun=Stampfun,
                                    stamps=Stamps}, SK) ->
 
+    NewThingToSign = case ThingToSign of
+                         undefined -> undefined;
+                         _ -> tpke_pubkey:deserialize_element(tpke_privkey:public_key(SK), ThingToSign)
+                     end,
+    #hbbft_data{secret_key=SK,
+                batch_size=BatchSize,
+                n=N,
+                f=F,
+                j=J,
+                round=Round,
+                buf=Buf,
+                max_buf=MaxBuf,
+                acs=hbbft_acs:deserialize(ACSData, SK),
+                acs_init=ACSInit,
+                sent_txns=SentTxns,
+                sent_sig=SentSig,
+                acs_results=ACSResults,
+                decrypted=Decrypted,
+                dec_shares=maps:map(fun(_, {Valid, Share}) ->
+                                            {Valid, hbbft_utils:binary_to_share(Share, SK)};
+                                       ({I, _}, Share) ->
+                                            %% compatability shim for old, untagged shares
+                                            {I, Enc} = lists:keyfind(I, 1, ACSResults),
+                                            EncKey = get_encrypted_key(SK, Enc),
+                                            DeserializedShare = hbbft_utils:binary_to_share(Share, SK),
+                                            Valid = tpke_pubkey:verify_share(tpke_privkey:public_key(SK), DeserializedShare, EncKey),
+                                            {Valid, DeserializedShare}
+                                    end, DecShares),
+                sig_shares=maps:map(fun(_, Share) -> hbbft_utils:binary_to_share(Share, SK) end, SigShares),
+                thingtosign=NewThingToSign,
+                stampfun=Stampfun,
+                stamps=Stamps};
+deserialize(M0, SK) ->
+    M = maps:map(fun(acs, V) -> V;
+                (_K, V) -> binary_to_term(V)
+                 end, M0),
+    #{batch_size := BatchSize,
```

Comment:
Is this too granular? maybe have more things bundled together (n, f, j, round, max_buf, acs_init, etc) together and only break out the 'bigger' things like buf, acs, decrypted, sig_shares, dec_shares, thingtosign?